README.rst

```diff
@@ -312,40 +312,72 @@ this concept, read the following section.
 Extending openwisp-firmware-upgrader
 ------------------------------------
 
-The ``tests/openwisp2/sample-firmware-upgrader`` may serve as an example for
-extending *openwisp-firmware-upgrader* in your own application.
+One of the core values of the OpenWISP project is `Software Reusability <http://openwisp.io/docs/general/values.html#software-reusability-means-long-term-sustainability>`_,
+for this reason *openwisp-firmware-upgrader* provides a set of base classes
+which can be imported, extended and reused to create derivative apps.
 
-*openwisp-firmware-upgrader* provides a set of models and admin classes which can
-be imported, extended and reused by third party apps.
+In order to implement your custom version of *openwisp-firmware-upgrader*,
+you need to perform the steps described in the rest of this section.
 
-To extend *openwisp-firmware-upgrader*, **you MUST NOT** add it to ``settings.INSTALLED_APPS``,
-but you must create your own app (which goes into ``settings.INSTALLED_APPS``), import the
-base classes from openwisp-firmware-upgrader and add your customizations.
-
-In order to help django find the static files and templates of *openwisp-firmware-upgrader*,
-you need to perform the steps described below.
+When in doubt, the code in the `test project <https://github.com/openwisp/openwisp-firmware-upgrader/tree/master/tests/openwisp2/>`_
```

Comment:
Also, what do you think about using relative path for these links? (`tests/openwisp2` in this case) 
So that these links work in fork and other branches as well, absolute path will work only in master branch!

Reply:
Would relative links from both type of URLs?

- https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/README.rst
- https://github.com/openwisp/openwisp-firmware-upgrader/

I think it would either work in one but not in the other.

Reply:
@nemesisdesign sorry, I didn't get that! :smile:

Reply:
The README can be viewed on two different URLs (the main repo URL and /blob/master/README.rst), which are on different levels, so using a relative path would work in one but not in the other.